### DISCUSSION TITLE
Extract data from .RFT

### DISCUSSION THREAD

**Discussion by ray63** [equinor/resdata#856] [Q&A] [answered]
What a fantastic library - Thanks guys - I can see this being way faster for certain analysis compared to PetrelRE - assuming one had all the licences available. 

I'd like to use pythion and extract RFT data to compare with measured data, however I am struggling to determine how to extract the data from the .RFT file.  I haven't found any examples and with limited python experience and its interaction with cpp , I haven't been able to decypher the library.

The best I've been able to investigate is a long the lines of code below.  There must be a simple way to extract a list of the data for a given well and date (be it RFT or PLT data), but I haven't worked it out hyet.

The next step will be to then load the trajectory data to enable a comparison of the measured data in MD, and compare to the modelled data - but that's the next problem (any hints or pointers to some examples here appreciated too)

<code>

from ecl.summary import EclSum
from ecl.rft import EclRFT ,  EclRFTCell, EclPLTCell

if __name__ == '__main__':

    case = sys.argv[1]
    rftFile = ecl.rft.EclRFTFile(case)    # load the RFT data
    rftHeaders = rftFile.get_headers()  # collection of (Wellname, date)

    print("Wells ", rftFile.get_num_wells())

    for i in range(0,len(rftFile)):
        print(rftFile.iget(i))

    for i in range(0,len(rftHeaders)):
        print(rftHeaders[i])

    print("Total number of RFTs : %d" % rftFile.size())
    print("RFTs matching M*    : %d" % rftFile.size(well="M*"))t
    print("RFTs at 01/01/2010   : %d" % rftFile.size(date=datetime.date(2010, 1, 1)))

    oneRFT= rftFile.get(well_name="MR6A-PH",date=datetime.date(2014, 6, 7))

</code>

**Comment by berland** [comment, marked as answer]
Hi, have a look at `ecl2df`, which wraps this `ecl` and provides a Pandas interface to parts of it, including the RFT support:

https://github.com/equinor/ecl2df
https://equinor.github.io/ecl2df/
https://equinor.github.io/ecl2df/usage/rft.html
https://github.com/equinor/ecl2df/blob/master/ecl2df/rft.py

**Comment by ray63** [comment]
Thanks berland - great link to more useful tools to investigate further.
